pkg/operator/target_config_reconciler.go

```diff
@@ -298,11 +325,32 @@ func (c TargetConfigReconciler) sync(ctx context.Context, syncCtx factory.SyncCo
 	}
 
 	for _, certificate := range certificateData {
-		err = c.manageCertificateCR(c.ctx, kueue, certificate.dnsNames, certificate.commonName, certificate.secretName, certificate.certificateName)
+		cert, _, err := c.manageCertificateCR(c.ctx, kueue, certificate.dnsNames, certificate.commonName, certificate.secretName, certificate.certificateName)
 		if err != nil {
 			klog.Errorf("unable to manage certificate err: %v", err)
 			return err
 		}
+		specAnnotations["certificate/"+cert.GetName()] = cert.GetResourceVersion()
+	}
+
+	// Wait for the webhook certificate to be ready before creating webhooks
+	// This prevents webhook timeout errors when the certificate isn't provisioned yet
+	err = cert.WaitForCertificateReady(c.ctx, c.dynamicClient, c.operatorNamespace, "webhook-cert", 2*time.Minute)
+	if err != nil {
+		klog.Warningf("Webhook certificate not ready yet: %v - will retry on next reconciliation", err)
+		return err
+	}
+
+	err = cert.WaitForCertificateReady(c.ctx, c.dynamicClient, c.operatorNamespace, "metrics-certs", 2*time.Minute)
+	if err != nil {
+		klog.Warningf("Webhook certificate not ready yet: %v - will retry on next reconciliation", err)
+		return err
+	}
+
+	err = cert.WaitForCertificateReady(c.ctx, c.dynamicClient, c.operatorNamespace, "kueue-visibility-server-cert", 2*time.Minute)
+	if err != nil {
+		klog.Warningf("Webhook certificate not ready yet: %v - will retry on next reconciliation", err)
```

Comment:
typo: kueue visibility server cert is being handled and not webhook certificate.